ios/ZingoMobile/AppDelegate.m

```diff
@@ -275,24 +346,37 @@ - (void)registerTasks {
 
 - (void)startBackgroundTask:(NSString *)noValue {
     NSLog(@"BGTask startBackgroundTask called");
+    RPCModule *rpcmodule = [RPCModule new];
     
     // Schedule tasks for the next time
     [self scheduleBackgroundTask];
     [self scheduleSchedulerBackgroundTask];
     
     if (!isConnectedToWifi) {
+        // save info in background json
+        NSTimeInterval timeStamp = [[NSDate date] timeIntervalSince1970];
+        // NSTimeInterval is defined as double
+        NSNumber *timeStampObj = [NSNumber numberWithDouble: timeStamp];
+        NSString *timeStampStr = [timeStampObj stringValue];
+        NSString *jsonBackgroud = [NSString stringWithFormat: @"%@%@%@%@%@%@%@", @"{\"batches\": \"", @"0", @"\", \"message\": \"", @"No wifi KO.", @"\", \"date\": \"", timeStampStr, @"\"}"];
+        [rpcmodule saveBackgroundFile:jsonBackgroud];
+      
         NSLog(@"BGTask startBackgroundTask: not connected to the wifi");
         [bgTask setTaskCompletedWithSuccess:NO];
         bgTask = nil;
         return;
     }
 
-    // I can check the time remaining here & make a choice
-    // when this `time remaing` is cracy big, something wrong is happening.
-    // in my testing this time is always something like 300 seconds. (the famous 5 min).
-    NSLog(@"BEFORE RUN TASKS - Time Remaining: %f", [[UIApplication sharedApplication] backgroundTimeRemaining]);
-    if ([[UIApplication sharedApplication] backgroundTimeRemaining] > 1000000000) {
-        NSLog(@"BGTask startBackgroundTask: time remainig TOO cracy high %f", [[UIApplication sharedApplication] backgroundTimeRemaining]);
+    if (!isCharging) {
+        // save info in background json
+        NSTimeInterval timeStamp = [[NSDate date] timeIntervalSince1970];
+        // NSTimeInterval is defined as double
+        NSNumber *timeStampObj = [NSNumber numberWithDouble: timeStamp];
+        NSString *timeStampStr = [timeStampObj stringValue];
+        NSString *jsonBackgroud = [NSString stringWithFormat: @"%@%@%@%@%@%@%@", @"{\"batches\": \"", @"0", @"\", \"message\": \"", @"No plug-in KO.", @"\", \"date\": \"", timeStampStr, @"\"}"];
```

Comment:
@LukasKorba and here I stored a simple message for the user/tester, to know why last night the BS didn't fire.... please tell me if I'm doing something wrong with this.